src/sources/util/http.rs

```diff
@@ -256,7 +262,21 @@ pub trait HttpSource: Clone + Send + Sync + 'static {
                                             error!(message = "Tried to send the following event.", %error);
                                             warp::reject::custom(RejectShuttingDown)
                                         })
-                                        .map_ok(|_| warp::reply())
+                                        .and_then(|_| async move {
+                                            let status = match receiver {
+                                                None => BatchStatus::Delivered,
+                                                Some(receiver) => receiver.await.unwrap_or(BatchStatus::Delivered),
+                                            };
+                                            match status {
+                                                BatchStatus::Delivered => Ok(warp::reply()),
+                                                BatchStatus::Failed => Err(warp::reject::custom(
+                                                    ErrorMessage::new(
+                                                        StatusCode::BAD_REQUEST,
```

Comment:
This should probably be a 5xx level code, but it wasn't clear which is most appropriate here.

Reply:
Yeah, ISE should lead to a retry. I am thinking this batch status needs to be divided from just `Failed` into `Rejected` and `Errored`, where the former would produce a 4xx code and the latter a 5xx code.